uploadservice/src/main/java/net/gotev/uploadservice/MultipartUploadTask.java

```diff
@@ -143,10 +145,13 @@ private void writeFiles(BodyWriter bodyWriter) throws IOException {
 
     @Override
     protected void onSuccessfulUpload() {
-        for (UploadFile file : params.getFiles()) {
-            addSuccessfullyUploadedFile(file.getPath());
+        // addSuccessfullyUploadedFile removes the file from the list so we have
+        // to use a copy to not get a java.util.ConcurrentModificationException
+        List<UploadFile> files = new ArrayList<>(params.getFiles());
```

Comment:
This was my first attempt but if we erase the element in the function addSuccessfullyUploadedFile, the iterator still becomes invalid. This was the code I tried, maybe I used the iterators wrong:

        Iterator<UploadFile> i = params.getFiles().iterator();
        while (i.hasNext()) {
            addSuccessfullyUploadedFile(i.next());
            // params.removeFile erases the element and the iterator is invalid. i.remove() would be correct but then, the method above must not delete the element
        }